src/analysis/schwartz_space.lean

```diff
@@ -403,6 +405,52 @@ begin
   rwa [pow_zero, one_mul] at this,
 end
 
+/-- The supremum of all seminorms `schwartz_map.seminorm 𝕜 k' n'` with `k' ≤ k` and `n' ≤ n`. -/
+@[protected]
+def seminorm_sup (k n : ℕ) : seminorm 𝕜 𝓢(E, F) :=
+((finset.range (k + 1)) ×ˢ (finset.range (n + 1))).sup (λ n, seminorm 𝕜 n.1 n.2)
+
+lemma le_seminorm_sup {k n k' n' : ℕ} (hk : k' ≤ k) (hn : n' ≤ n) :
+  (seminorm 𝕜 k' n' : seminorm 𝕜 𝓢(E, F)) ≤ seminorm_sup 𝕜 k n :=
+begin
+  have : (k', n') ∈ (finset.range (k + 1)).product (finset.range (n + 1)) :=
+  begin
+    simp only [finset.mem_range, finset.mem_product],
+    exact ⟨nat.lt_succ_of_le hk, nat.lt_succ_of_le hn⟩,
+  end,
+  exact @finset.le_sup _ _ _ _ _ (λ (n : ℕ × ℕ), seminorm 𝕜 n.1 n.2) _ this,
+end
```

Comment:
```suggestion
finset.le_sup $ finset.mem_Iic.2 ⟨hk, hn⟩ 
```

Reply:
I did not get this to work, Lean needs some help to unify the statement.

Reply:
What about
```suggestion
@finset.le_sup $ finset.mem_Iic.2 ⟨hk, hn⟩ 
```

Reply:
````
type mismatch at application
  ⟨⁇, hn⟩
term
  hn
has type
  n' ≤ n
but is expected to have type
  ?m_1.snd ≤ ?m_2.snd
````

Reply:
Last try:
```suggestion
@finset.le_sup _ _ _ _ _ (λ n : ℕ × ℕ, seminorm 𝕜 n.1 n.2) _ $ finset.mem_Iic.2 $ prod.mk_le_mk.2 ⟨hk, hn⟩ 
```

Reply:
This works, thank you.